### PR TITLE
objc: add autogenerated header to generated files

### PR DESCRIPTION
**Pull request by jagobagascon**
Currently the code generated by the gRPC Objective-C plugin does not have any indication that it is an autogenerated file. The rest of the plugins do this by adding a comment at the beginning of the file. I think that Objective-C plugin should behave the same way to keep consistency, also having that header may be useful for developer in certain situations.

@markdroth

### CODE REVIEW

File: src/compiler/objective_c_plugin.cc [hunk 1]
```diff
@@ -127,6 +127,10 @@ class ObjectiveCGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
       }
     }
 
+    // Write out a file header.
+    ::grpc::string file_header = "// Code generated by gRPC proto compiler.  DO NOT EDIT!\n"
+        "// source: " + file->name() + "\n";
```
Comment (muxi) [inline src/compiler/objective_c_plugin.cc:132]: Let's add an additional "\n" at the end for pretty

**Comment by muxi** [comment]
Thanks for the PR.

File: src/compiler/objective_c_plugin.cc (hunk 1 above)
Comment (jagobagascon) [reply src/compiler/objective_c_plugin.cc]: Done!

**Comment by stanley-cheung** [comment]
I believe there are some formatting changes required by clang_tidy

https://source.cloud.google.com/results/invocations/6d8c5436-f762-4d7b-a53d-489e6d4e1356/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_sanity/log

**Comment by jagobagascon** [comment]
> I believe there are some formatting changes required by clang_tidy
> 
> https://source.cloud.google.com/results/invocations/6d8c5436-f762-4d7b-a53d-489e6d4e1356/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_sanity/log

Should be fixed now

**Comment by jagobagascon** [comment]
Some checks are failing but the errors don't seem to be related to changes in this PR.

**Comment by muxi** [comment]
@jagobagascon Agreed. Could you sign the CLA please?

**Pull request merged**